scripts/prepare_release.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+#
+# This script:
+#   1. parses the version number from the branch name
+#   2. updates version.py files to match that version
+#   3. iterates through CHANGELOG.md files and updates any files containing
+#      unreleased changes
+#   4. sets the output variable 'version_updated' to determine whether
+#      the github action to create a pull request should run. this allows
+#      maintainers to merge changes back into the release branch without
+#      trigerring unnecessary pull requests
+#
+
+VERSION=`echo $1 | awk -F "/" '{print $NF}'`
+echo "Using version ${VERSION}"
+
+# check the version matches expected versioning e.g
+# 0.6, 0.6b, 0.6b0, 0.6.0
+if [[ ! "${VERSION}" =~ ^([0-9])(\.*[0-9]{1,5}[a-b]*){1,3}$ ]]; then
+    echo "Version number invalid: $VERSION"
+    exit 1
+fi
+
+function update_version_file() {
+    errors=0
+    for f in `find . -name version.py`; do
+        # check if version is already in version.py
+        grep -q ${VERSION} $f;
+        rc=$?
+        if [ $rc == 0 ]; then
+            errors=1
+            echo "${f} already contains ${VERSION}"
+            continue
+        fi
+        # update version.py
+        perl -i -pe "s/__version__.*/__version__ = \"${VERSION}\"/g" ${f};
+        git add ${f};
+        echo "Updating ${f}"
+    done
+    if [ ${errors} != 0 ]; then
+        echo "::set-output name=version_updated::0"
```

Comment:
just out of curiosity, does the version_updated variable even need to be created? I guess I'm wondering if we run into an issue that is not handled appropriately, then this statement won't be echoed, thus not setting the output. 

It's more robust to just check for lack of existence in these scenarios.